src/components/Modal/types.ts

```diff
@@ -0,0 +1,30 @@
+import { TPAComponentProps } from '../../types';
+import { RefObject } from 'react';
+
+export interface ModalProps extends TPAComponentProps {
+  isOpen: boolean;
+  rootElement: HTMLElement;
+  onClose?(): void;
+  contentClassName?: string;
+  closeOnClickOutside?: boolean;
+  minHeight?: string;
+  maxHeight?: string;
+  minWidth?: string;
+  maxWidth?: string;
+  closeButtonRef?: RefObject<any>;
```

Comment:
So these should probably not be part of the `Modal`'s proptypes, but in the `Dialog`'s